pkg/yurtctl/cmd/convert/edgenode.go

```diff
@@ -0,0 +1,308 @@
+/*
+Copyright 2021 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package convert
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+
+	"github.com/alibaba/openyurt/pkg/projectinfo"
+	enutil "github.com/alibaba/openyurt/pkg/yurtctl/util/edgenode"
+	kubeutil "github.com/alibaba/openyurt/pkg/yurtctl/util/kubernetes"
+	"github.com/alibaba/openyurt/pkg/yurthub/healthchecker"
+)
+
+const (
+	kubeletConfigRegularExpression = "\\-\\-kubeconfig=.*kubelet.conf"
+	apiserverAddrRegularExpression = "server: (http(s)?:\\/\\/)?[\\w][\\w]{0,62}(\\.[\\w][-\\w]{0,62})+(:[\\d]{1,5})?"
+	hubHealthzCheckFrequency       = 10 * time.Second
+	failedRetry                    = 5
+	filemode                       = 0666
+	dirmode                        = 0755
+)
+
+// ConvertEdgeNodeOptions has the information required by sub command convert edgenode
+type ConvertEdgeNodeOptions struct {
+	clientSet       *kubernetes.Clientset
+	YurthubImage    string
+	JoinToken       string
+	PodMainfestPath string
+	kubeletSvcPath  string
+	openyurtDir     string
+}
+
+// NewConvertEdgeNodeOptions creates a new ConvertEdgeNodeOptions
+func NewConvertEdgeNodeOptions() *ConvertEdgeNodeOptions {
+	return &ConvertEdgeNodeOptions{}
+}
+
+// NewConvertEdgeNodeCmd generates a new sub command convert edgenode
+func NewConvertEdgeNodeCmd() *cobra.Command {
+	c := NewConvertEdgeNodeOptions()
+	cmd := &cobra.Command{
+		Use:   "edgenode",
```

Comment:
i think we should specify node name for this subcommand, like `yurtctl convert/revert edgenode node-name --xxx=xxx`,  so this subcommand can be worked at everywhere with kubeconfig.

Reply:
And this subcommand can work only on the edge node that user want to convert or revert in current design, I think it's not user-friendly.

Reply:
so this sub command can need run servant pod to exec convert or revert command.